cpp/include/cudf/strings/strings_column_handler.hpp

```diff
@@ -0,0 +1,232 @@
+      name=2     ///< sort by characters code-points
+  };
+
+private:
+  const column_view _parent;
+  rmm::mr::device_memory_resource* _mr;
+
+};
+
+namespace strings
+{
+
+// array.cu
+/**---------------------------------------------------------------------------*
+ * @brief Returns a new strings column created from a subset of
+ * of this instance's strings column.
+ *
+ * @code
+ * s1 = ["a", "b", "c", "d", "e", "f"]
+ * s2 = sublist( s1, 2 )
+ * s2 is ["c", "d", "e", "f"]
+ * @endcode
+ * 
+ * @param start Index of first string to use.
+ * @param end Index of last string to use.
+ *        Default -1 indicates the last element.
+ * @param step Increment value between indexes.
+ *        Default step is 1.
+ * @param stream CUDA stream to use kernels in this method.
+ * @return New strings column of size (end-start)/step.
+ *---------------------------------------------------------------------------**/
+std::unique_ptr<cudf::column> sublist( strings_column_handler handler,
+                                       size_type start, size_type end=-1,
+                                       size_type step=1,
+                                       cudaStream_t stream=(cudaStream_t)0 );
+
+/**---------------------------------------------------------------------------*
+ * @brief Returns a new strings column created this strings instance using
+ * the specified indices to select the strings.
+ * 
+ * @code
+ * s1 = ["a", "b", "c", "d", "e", "f"]
+ * map = [0, 2]
+ * s2 = gather( s1, map )
+ * s2 is ["a", "c"]
+ * @endcode
+ *
+ * @param gather_map The indices with which to select strings for the new column.
+ *        Values must be within [0,size()) range.
+ * @param stream CUDA stream to use kernels in this method.
+ * @return New strings column of size indices.size()
+ *---------------------------------------------------------------------------**/
+std::unique_ptr<cudf::column> gather( strings_column_handler handler,
+                                      cudf::column_view gather_map,
+                                      cudaStream_t stream=(cudaStream_t)0 );
+
+/**---------------------------------------------------------------------------*
+ * @brief Returns a new strings column that is a sorted version of the
+ * strings in this instance.
+ *
+ * @param stype Specify what attribute of the string to sort on.
+ * @param ascending Sort strings in ascending or descending order.
+ * @param nullfirst Sort nulls to the beginning or the end of the new column.
+ * @param stream CUDA stream to use kernels in this method.
+ * @return New strings column with sorted elements of this instance.
+ *---------------------------------------------------------------------------**/
+std::unique_ptr<cudf::column> sort( strings_column_handler handler,
+                                    strings_column_handler::sort_type stype,
+                                    bool ascending=true,
+                                    bool nullfirst=true,
```

Comment:
We should be consistent in sorting functions with how we specify ascending and null ordering. 

Can you use the enums I defined here (I'll move them to `types.hpp` so you don't have to include `sorting.hpp`):
https://github.com/rapidsai/cudf/pull/2207/files?file-filters%5B%5D=#diff-57f3e3df872ff63b9685bc3a15f19ad1R39
https://github.com/rapidsai/cudf/pull/2207/files?file-filters%5B%5D=#diff-57f3e3df872ff63b9685bc3a15f19ad1R30